### PR TITLE
Add tags and patterns to template creation

### PR DESCRIPTION
**Pull request by polubis**

### CODE REVIEW

File: src/core/patterns/index.ts [hunk 1]
```diff
@@ -0,0 +1,4 @@
+import PatternsProvider from './PatternsProvider';
+
+export * from './PatternsProvider';
```
Comment (marcinreszko) [inline src/core/patterns/index.ts:3]: Enter

File: src/core/technologies/index.ts [hunk 2]
```diff
@@ -1,6 +1,4 @@
 import TechnologiesProvider from './TechnologiesProvider';
 
-export * from './models';
-export * from './utils';
-
+export * from './TechnologiesProvider';
```
Comment (marcinreszko) [inline src/core/technologies/index.ts:3]: Enter

File: src/modules/main/Main.tsx [hunk 3]
```diff
@@ -15,7 +15,7 @@ const TemplateDetails = withLazy(() => import('src/modules/template-details'));
 
 const TemplateDocumentation = withLazy(() => import('src/modules/template-documentation'));
 
-const TemplateCreation = withLazy(() => import('src/modules/template-creation'));
+const TemplateManagement = withLazy(() => import('src/modules/template-management'));
 
 interface MainProps extends RouteChildrenProps {}
```
Comment (marcinreszko) [inline src/modules/main/Main.tsx:21]: Namespace?

File: src/modules/template-management/template-form/useTemplateManagement.ts [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+import { addTemplate, AddTemplatePayload } from 'core/api';
+
+interface State {
+  pending: boolean;
+  error: string;
+  id: string | null;
+}
+
+type Return = [State, (payload: AddTemplatePayload) => Promise<void>];
+
```
Comment (marcinreszko) [inline src/modules/template-management/template-form/useTemplateManagement.ts:13]: Namespace aswell?

Comment (polubis) [reply src/modules/template-management/template-form/useTemplateManagement.ts:13]: for useTemplateManagement i need to create namespace - useTemplateManagement which is invalid if you look from ts good practices perspective. Every model should have name starting from big letter - so for hooks we can't use namespace

File: src/core/technologies/index.ts (hunk 2 above)
Comment (polubis) [reply src/core/technologies/index.ts:3]: Done

File: src/core/patterns/index.ts (hunk 1 above)
Comment (polubis) [reply src/core/patterns/index.ts:3]: done

File: src/modules/main/Main.tsx (hunk 3 above)
Comment (polubis) [reply src/modules/main/Main.tsx:21]: Done

**Pull request merged**